api/grpcserver/debug_service.go

```diff
@@ -65,3 +69,61 @@ func (d DebugService) Accounts(_ context.Context, in *empty.Empty) (*pb.Accounts
 func (d DebugService) NetworkInfo(ctx context.Context, _ *empty.Empty) (*pb.NetworkInfoResponse, error) {
 	return &pb.NetworkInfoResponse{Id: d.identity.ID().String()}, nil
 }
+
+// ProposalsStream streams all proposals confirmed by hare.
+func (d DebugService) ProposalsStream(_ *emptypb.Empty, stream pb.DebugService_ProposalsStreamServer) error {
+	var (
+		eventch <-chan interface{}
+		fullch  <-chan struct{}
+	)
+	if sub := events.SubcribeProposals(); sub != nil {
+		eventch, fullch = consumeEvents(stream.Context(), sub)
+	}
+	if eventch == nil || fullch == nil {
+		return status.Errorf(codes.FailedPrecondition, "event reporting is not enabled")
+	}
```

Comment:
`consumeEvents` always returns non-nil channels, so `codes.FailedPrecondition` would mean that `events.SubcribeProposals() == nil`, maybe we can simplify this?
```suggestion
	sub := events.SubcribeProposals()
	if sub == nil {
		return status.Errorf(codes.FailedPrecondition, "event reporting is not enabled")
	}
	eventch, fullch := consumeEvents(stream.Context(), sub)
```

Reply:
thanks, applied